core/vm/ovm_state_manager.go

```diff
@@ -0,0 +1,125 @@
+package vm
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type stateManagerFunction func(*EVM, *Contract, map[string]interface{}) ([]interface{}, error)
+
+var funcs = map[string]stateManagerFunction{
+	"owner":                                    owner,
+	"setAccountNonce":                          setAccountNonce,
+	"getAccountNonce":                          getAccountNonce,
+	"getAccountEthAddress":                     getAccountEthAddress,
+	"getContractStorage":                       getContractStorage,
+	"putContractStorage":                       putContractStorage,
+	"isAuthenticated":                          nativeFunctionTrue,
+	"hasAccount":                               nativeFunctionTrue,
+	"hasEmptyAccount":                          nativeFunctionTrue,
+	"hasContractStorage":                       nativeFunctionTrue,
+	"testAndSetAccountLoaded":                  nativeFunctionTrue,
+	"testAndSetAccountChanged":                 nativeFunctionTrue,
+	"testAndSetContractStorageLoaded":          nativeFunctionTrue,
+	"testAndSetContractStorageChanged":         nativeFunctionTrue,
+	"incrementTotalUncommittedAccounts":        nativeFunctionVoid,
+	"incrementTotalUncommittedContractStorage": nativeFunctionVoid,
+	"initPendingAccount":                       nativeFunctionVoid,
+	"commitPendingAccount":                     nativeFunctionVoid,
+}
+
+func callStateManager(input []byte, evm *EVM, contract *Contract) (ret []byte, err error) {
+	rawabi := OvmStateManager.ABI
+	abi := &rawabi
+
+	method, err := abi.MethodById(input)
+	if err != nil {
+		return nil, err
+	}
+
+	var inputArgs = make(map[string]interface{})
+	err = method.Inputs.UnpackIntoMap(inputArgs, input[4:])
+	if err != nil {
+		return nil, err
+	}
+
+	fn, exist := funcs[method.RawName]
+	if !exist {
+		return nil, fmt.Errorf("Native OVM_StateManager function not found for method '%s'", method.RawName)
+	}
+
+	outputArgs, err := fn(evm, contract, inputArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	returndata, err := method.Outputs.PackValues(outputArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	return returndata, nil
+}
+
+func owner(evm *EVM, contract *Contract, args map[string]interface{}) ([]interface{}, error) {
+	origin := evm.Context.Origin
+
+	return []interface{}{origin}, nil
+}
+
+func setAccountNonce(evm *EVM, contract *Contract, args map[string]interface{}) ([]interface{}, error) {
+	address := args["_address"].(common.Address)
```

Comment:
```suggestion
	address, ok := args["_address"].(common.Address)
```

I wonder if a client may be able to provide "bad arguments" in a way and cause a panic here? May be better to handle the error (same for all other interface -> type casts)

Reply:
I agree and think its a better idea to be thorough with the error checking